test/upload_test.sh

```diff
@@ -50,13 +52,20 @@ setup_network_conditions() {
             "sudo pfctl -e || true"
         )
     else
-        # TODO implement commands for linux
-        return 0
+        bw_in_bits_per_sec="${bw%/s}"
+        bdp_bytes=$(echo "scale=2; $bdp_bits / 8" | bc)
+        bdp_bytes_rounded_up=$(LC_NUMERIC=C printf "%.0f" "$bdp_bytes")
+        plr_p=$(echo "scale=4; $plr * 100" | bc)
+        plr_p=$(LC_NUMERIC=C printf "%.2f" "$plr_p")
+        set_condition_commands=(
+            "sudo tc qdisc add dev lo root handle 1: tbf rate $bw_in_bits_per_sec burst $bdp_bytes_rounded_up limit 30000"
+            "sudo tc qdisc add dev lo parent 1:1 handle 10: netem delay ${delay_ms}ms loss ${plr_p}%"
+        )
     fi
 
     for command in "${set_condition_commands[@]}"; do
-        echo $command
-        echo $root_password | sudo -S bash -c "$command"
+        echo "$command"
+        bash -c "$command"
```

Comment:
Shouldn't this use `sudo`? Also, couldn't you use `eval` rather than starting another shell?

Reply:
Yes, I can use `eval $command`, but not `sudo eval $command`.
I got this warning from `shellcheck`:
```
Can't use sudo with builtins like eval. Did you want sudo sh -c .. instead?shellcheckSC2232
```